sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientRetryPolicy.java

```diff
@@ -374,6 +391,24 @@ CosmosDiagnostics getCosmosDiagnostics() {
         return cosmosDiagnostics;
     }
 
+    private boolean shouldRetryWriteOnServiceUnavailable(
+        boolean nonIdempotentWriteRetriesEnabled,
+        boolean isWebExceptionRetriable,
+        CosmosException cosmosException) {
+
+        if (nonIdempotentWriteRetriesEnabled || isWebExceptionRetriable) {
+            return true;
+        }
+
+        if (cosmosException instanceof ServiceUnavailableException) {
+            ServiceUnavailableException serviceUnavailableException = (ServiceUnavailableException) cosmosException;
+            return serviceUnavailableException.isBasedOn503ResponseFromService()
+                || serviceUnavailableException.getStatusCode() == HttpConstants.SubStatusCodes.SERVER_GENERATED_410;
```

Comment:
Here we are comparing `statusCode` with `subStatusCode` - this would never be satisfied, right?

Reply:
haha good catch thanks @jeet1995